core/controllers/reader.py

```diff
@@ -111,17 +111,53 @@ def get(self, exploration_id):
 class ExplorationPage(base.BaseHandler):
     """Page describing a single exploration."""
 
+    URL_PATH_ARGS_SCHEMAS = {
+        'exploration_id': {
+            'schema': {
+                'type': 'basestring'
+            }
+        }
+    }
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {
+            'v': {
+                'schema': {
+                    'type': 'int'
+                },
+                'default_value': None
+            },
+            'parent': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
+            },
+            'iframed': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
+            },
+            'collection_id': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
+            }
+        }
+    }
+
     @acl_decorators.can_play_exploration
     def get(self, exploration_id):
         """Handles GET requests.
 
         Args:
             exploration_id: str. The ID of the exploration.
         """
-        version_str = self.request.get('v')
+        version_str = self.normalized_request.get('v')
         version = int(version_str) if version_str else None
```

Comment:
Not required since the normalized value should already be int.

Reply:
It is handled in such a way that if version is not convertable into int then it is assigned to None, but in case of SVS architecture if version is not convertable into 'int' it will raise 400 error.